tools/cuds-prioritized-attribution/composer/main.py

```diff
@@ -0,0 +1,225 @@
+        """Executes first query to break out lines into project and credits.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary of string key-value pairs of environment vars.
+
+        Returns:
+             None
+        """
+        execute_query(bq_client,
+                      env_vars,
+                      project_label_credit_data,
+                      env_vars['project_label_credit_breakout_table'],
+                      False)
+
+    def distribute_commitments(bq_client: bigquery.Client,
+        env_vars: Dict[str, Union[str, bool]]) -> None:
+        """Executes second query to compute commitments per SKU.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary of string key-value pairs of environment vars.
+
+        Returns:
+             None
+        """
+        execute_query(bq_client,
+                      env_vars,
+                      distribute_commitment,
+                      env_vars['distribute_commitments_table'],
+                      False)
+
+    def billing_output(bq_client: bigquery.Client,
+        env_vars: Dict[str, Union[str, bool]]) -> None:
+        """Executes third query to format output schema.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary of string key-value pairs of environment vars.
+
+        Returns:
+             None
+        """
+        execute_query(bq_client,
+                      env_vars,
+                      billingoutput,
+                      env_vars['corrected_table_name'],
+                      True)
+
+    def delete_temp_tables(bq_client: bigquery.Client,
+        env_vars: Dict[str, Union[str, bool]]) -> None:
+        """Deletes the three temporary tables that were created by the DAG.
+
+        Args:
+            bq_client: bigquery.Client object
+            env_vars: Dictionary holding key-value pair of environment vars.
+
+        Returns:
+            None
+        """
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['temp_commitments_table_name'])
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['distribute_commitments_table'])
+        helper_function.delete_table(bq_client,
+                                     env_vars['corrected_dataset_id'],
+                                     env_vars['project_label_credit_breakout_table'])
+
+    # Obtain values for all of the environment variables
+    KEY_LIST = ['project_id', 'billing_export_table_name',
+                'corrected_dataset_id', 'corrected_table_name',
+                'commitments_table_name', 'enable_cud_cost_attribution', 'cud_cost_attribution_option']
+    ENV_VARS = get_env_variables(KEY_LIST)
```

Comment:
Why are we using environment variables instead of [Airflow Variables](https://airflow.apache.org/docs/stable/concepts.html#variables)?

I fear this will not work in distributed airflow environments and sets a bad example.

Reply:
bump.

Reply:
I had investigated this initially, but didn't think it was the cleanest approach in composer. Because you can only set airflow variables at the run environment stage rather than create environment. Given that it can take ~40 min to create the environment, I was worried that users would forget to do the update CLI command for setting env variables.

Reply:
defer to a future PR (maybe)